### PR TITLE
[FIX] mail: send is_main with the attachments

### PR DESCRIPTION
**Pull request by phenix-factory**
task-2783076

### CODE REVIEW

File: addons/mail/static/src/models/thread/thread.js [hunk 1]
```diff
@@ -670,24 +670,41 @@ function factory(dependencies) {
...
+            }, { shadow: true });
+            const values = {};
+            if (attachmentsData) {
+                Object.assign(values, {
+                    areAttachmentsLoaded: true,
+                    isLoadingAttachments: false,
+                    originThreadAttachments: [['insert-and-replace', attachmentsData]],
+                });
+            }
+            this.update(values);
```
Comment (alexkuhn) [inline addons/mail/static/src/models/thread/thread.js:707]: This is partial backport of code. Could you isolate the introduction to `fetchData` without the `is_main` fix, and then make a following commit for fixing specifically the `is_main`?

Comment (alexkuhn) [reply addons/mail/static/src/models/thread/thread.js:707]: It's ok then, it's a fix anyway. Lots of diff, but if it looks similar than master then it's quite safe.

**Review by seb-odoo — CHANGES_REQUESTED** [review]

File: addons/mail/controllers/main.py [hunk 2]
```diff
@@ -294,3 +294,11 @@ def message_get_suggested_recipients(self, model, res_ids):
         except:
             return {}
         return records._message_get_suggested_recipients()
+
+    @http.route('/mail/thread/data', methods=['POST'], type='json', auth='user')
+    def mail_thread_data(self, thread_model, thread_id, request_list, **kwargs):
```
[the remark this answers is not in the document]
Comment (phenix-factory) [reply addons/mail/controllers/main.py:299]: Create a new file in v14 just for a route doesn't seams right. This route is not a lot of code;

File: addons/mail/static/tests/helpers/mock_server.js [hunk 3]
```diff
@@ -576,7 +631,8 @@ MockServer.include({
     _mockMailChannel_channelChannelNotifications(ids, partner_ids) {
         const notifications = [];
         for (const partner_id of partner_ids) {
-            const user = this._getRecords('res.users', [['partner_id', 'in', partner_id]])[0];
+
+    const user = this._getRecords('res.users', [['partner_id', 'in', partner_id]])[0];
```
Comment (seb-odoo) [inline addons/mail/static/tests/helpers/mock_server.js:635]: bad diff

File: addons/mail/controllers/main.py [hunk 4]
```diff
@@ -87,7 +87,7 @@ def _redirect_to_record(cls, model, res_id, access_token=None, **kwargs):
                     #   _get_mail_redirect_suggested_company
                     #   - If no company, then redirect to the messaging
                     #   - Merge the suggested company with the companies on the cookie
-                    # - Make a new access test if it succeeds, redirect to the record. Otherwise, 
+                    # - Make a new access test if it succeeds, redirect to the record. Otherwise,
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/controllers/main.py:90]: comment still open

File: addons/mail/models/ir_attachment.py [hunk 5]
```diff
@@ -34,3 +35,29 @@ def register_as_main_attachment(self, force=True):
                     related_record.message_main_attachment_id = self
                 except AccessError:
                     pass
+
+    def _attachment_format(self, commands=False):
```
Comment (seb-odoo) [inline addons/mail/models/ir_attachment.py:39]: use the method whenever possible (at least message format?)

File: addons/mail/static/src/models/attachment/attachment.js [hunk 6]
```diff
@@ -46,7 +49,6 @@ function factory(dependencies) {
                     model: data.res_model,
                 }]];
             }
-
```
Comment (seb-odoo) [inline addons/mail/static/src/models/attachment/attachment.js:49]: (unrelated diff)

File: addons/mail/models/ir_attachment.py [hunk 7]
```diff
@@ -34,3 +35,29 @@ def register_as_main_attachment(self, force=True):
...
+        safari = request and request.httprequest.user_agent and request.httprequest.user_agent.browser == 'safari'
+        res_list = []
+        for attachment in self:
+            res = {
+                'id': attachment.id,
+                'filename': attachment.name,
+                'name': attachment.name,
+                'mimetype': 'application/octet-stream' if safari and attachment.mimetype and 'video' in attachment.mimetype else attachment.mimetype,
```
Comment (seb-odoo) [inline addons/mail/models/ir_attachment.py:47]: missing `checksum` key

File: addons/mail/static/tests/helpers/mock_server.js (hunk 3 above)
Comment (seb-odoo) [reply addons/mail/static/tests/helpers/mock_server.js:636]: comment still open

**Comment by seb-odoo** [comment]
runbot issue (also missing checksum field in model) + unsync community/enterprise

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
